frame/nis/src/lib.rs

```diff
@@ -615,19 +625,21 @@ pub mod pallet {
 			ensure!(queue_index < queue_count, Error::<T>::DurationTooBig);
 
 			let bid = Bid { amount, who };
-			let new_len = Queues::<T>::try_mutate(duration, |q| -> Result<u32, DispatchError> {
-				let pos = q.iter().position(|i| i == &bid).ok_or(Error::<T>::UnknownBid)?;
-				q.remove(pos);
-				Ok(q.len() as u32)
-			})?;
 
+			let mut queue = Queues::<T>::get(duration);
+			let pos = queue.iter().position(|i| i == &bid).ok_or(Error::<T>::UnknownBid)?;
+			queue.remove(pos);
```

Comment:
This is a great example of a code that's being rewritten simpler using transactional storage in mind @liamahoran